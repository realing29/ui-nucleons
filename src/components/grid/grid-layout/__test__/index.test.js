import React from 'react';
import { shallow } from 'enzyme/build';
import GridLayout from '../';

describe('<GridLayout />', () => {
  it('should renders correctly without props', () => {
    const wrapper = shallow(
      <GridLayout>
        <div className='test-content'>123</div>
      </GridLayout>
    );
    expect(wrapper.find('.grid-layout')).toHaveLength(1);
    expect(wrapper.find('.test-content')).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });
  it('should renders correctly with props', () => {
    const wrapper = shallow(
      <GridLayout
        containerTag='section'
        containerProps={{
          className: 'section-class',
          name: 'some-name',
        }}
      >
        <div className='test'>some content</div>
      </GridLayout>
    );
    expect(wrapper.find('section.grid-layout')).toHaveLength(1);
    expect(wrapper.find('section.section-class')).toHaveLength(1);
    expect(wrapper.find('.test')).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });
  it('should set correct classes with custom padding and maxWidth props', () => {
    const wrapper = shallow(
      <GridLayout
        xxlPadding='sm'
        xlPadding='sm'
        lgPadding='sm'
        mdPadding='sm'
        smPadding='sm'
        xsPadding='sm'
        xxsPadding='sm'
        xxxsPadding='sm'
        xxlMaxWidth='sm'
        xlMaxWidth='sm'
        lgMaxWidth='sm'
        mdMaxWidth='sm'
        smMaxWidth='sm'
        xsMaxWidth='sm'
        xxsMaxWidth='sm'
        xxxsMaxWidth='sm'
      >
        <div className='test'>some content</div>
      </GridLayout>
    );
    expect(wrapper.find('.grid-layout-wrapper').prop('className')).toEqual(
      [
        'grid-layout-wrapper',
        'layout-xxl-padding-sm',
        'layout-xl-padding-sm',
        'layout-lg-padding-sm',
        'layout-md-padding-sm',
        'layout-sm-padding-sm',
        'layout-xs-padding-sm',
        'layout-xxs-padding-sm',
        'layout-xxxs-padding-sm',
      ].join(' ')
    );
    expect(wrapper.find('.grid-layout').prop('className')).toEqual(
      [
        'grid-layout',
        'layout-xxl-width-sm',
        'layout-xl-width-sm',
        'layout-lg-width-sm',
        'layout-md-width-sm',
        'layout-sm-width-sm',
        'layout-xs-width-sm',
        'layout-xxs-width-sm',
        'layout-xxxs-width-sm',
      ].join(' ')
    );
  });
});
