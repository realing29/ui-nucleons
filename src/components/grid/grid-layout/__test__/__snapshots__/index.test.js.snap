// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridLayout /> should renders correctly with props 1`] = `
<div
  className="grid-layout-wrapper layout-xxl-padding-sm layout-xl-padding-sm layout-lg-padding-sm layout-md-padding-lg layout-sm-padding-sm layout-xs-padding-sm layout-xxs-padding-md layout-xxxs-padding-sm"
>
  <section
    className="grid-layout layout-xxl-width-xl layout-xl-width-lg layout-lg-width-md layout-md-width-md layout-sm-width-sm layout-xs-width-xs layout-xxs-width-xs layout-xxxs-width-xs section-class"
    name="some-name"
  >
    <div
      className="test"
    >
      some content
    </div>
  </section>
</div>
`;

exports[`<GridLayout /> should renders correctly without props 1`] = `
<div
  className="grid-layout-wrapper layout-xxl-padding-sm layout-xl-padding-sm layout-lg-padding-sm layout-md-padding-lg layout-sm-padding-sm layout-xs-padding-sm layout-xxs-padding-md layout-xxxs-padding-sm"
>
  <div
    className="grid-layout layout-xxl-width-xl layout-xl-width-lg layout-lg-width-md layout-md-width-md layout-sm-width-sm layout-xs-width-xs layout-xxs-width-xs layout-xxxs-width-xs"
  >
    <div
      className="test-content"
    >
      123
    </div>
  </div>
</div>
`;
